### PR TITLE
Update Options to hide traceparent and tracestate

### PR DESCRIPTION
**Pull request by azabbasi**
This change will render certain options look empty to users so we are removing them from the API signatures.

We have the option to just mark some of the options internal and do not implement any overrides in customized classes, but that will result in TraceParent and TraceState have the incorrect naming format : Tracestate, Traceparent

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/CreateModelsOptions.cs [hunk 1]
```diff
@@ -7,16 +7,22 @@ namespace Azure.DigitalTwins.Core
...
     [CodeGenModel("DigitalTwinModelsAddOptions")]
-    public partial class CreateModelsOptions
+    internal partial class CreateModelsOptions
     {
-        // This class declaration changes the namespace; do not remove.
+        // This class declaration changes the namespace, class name and property visibility; do not remove.
+
+        // This class contains two properties (TraceParent ,TraceState) that are not intended to be used by the Track 2 SDKs.
+        // Marking these properties as internal.
+        #region internalProperties
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/CreateModelsOptions.cs:16]: We don't need regions to separate out properties by visibility spec, do we?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs [hunk 2]
```diff
@@ -1508,13 +1487,11 @@ async Task<Page<string>> FirstPageFunc(int? pageSizeHint)
...
                     };
 
-                    if (options == null)
-                    {
-                        options = new QueryOptions();
-                    }
-
                     // Page size hint is only specified by AsPages() methods, so we add it here manually since the user can't set it on the options object directly
-                    options.MaxItemsPerPage = pageSizeHint;
+                    QueryOptions options = new QueryOptions
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1491]: var

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs [hunk 3]
```diff
@@ -1537,13 +1514,11 @@ async Task<Page<string>> NextPageFunc(string nextLink, int? pageSizeHint)
...
```
Comment (drwill-ms) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1518]: var

**Review by drwill-ms — APPROVED** [review]
I really don't like the regions for internal properties, but other than that, looks good.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/CreateModelsOptions.cs (hunk 1 above)
Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/Customized/Models/CreateModelsOptions.cs:16]: no worries ! I can certainly remove the regions ... I don't feel that strongly about them.

**Comment by tg-msft** [comment]
If you're making the Options classes internal, do they get set as headers by the generated code?  Does that conflict with https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs?  // cc @pakrym as the expert here

**Comment by azabbasi** [comment]
> If you're making the Options classes internal, do they get set as headers by the generated code? Does that conflict with https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs? // cc @pakrym as the expert here

We were advised by the SDK team to not use TraceParent and TraceState in our API options. They will still be set but we won't get them as input from the user.

**Comment by tg-msft** [comment]
Yeah, that was my advice because the `RequestActivityPolicy` sets them automatically.  However [it won't replace them if already set](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs#L123).  I just want to double check [the generated code](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/digitaltwins/Azure.DigitalTwins.Core/src/Generated/DigitalTwinModelsRestClient.cs#L58) doesn't cause a problem.  If it does, we could also just yank all the traceparent/tracestate headers from your swagger with a README transform in our repo.

**Comment by timtay-microsoft** [comment]
>  it won't replace them if already set.

We shouldn't be passing down any values for these to the generated code, so we should be fine here

**Comment by pakrym** [comment]
FYI https://github.com/Azure/azure-sdk-for-net/pull/16318#is coming

**Pull request merged**